packages/captp/lib/sync.js

```diff
@@ -0,0 +1,86 @@
+// Lifted mostly from `@agoric/eventual-send/src/E.js`.
+
+import './types';
+
+/**
+ * Default implementation of GetApplySync.
+ *
+ * @type {GetApplySync}
+ */
+export const nearGetApplySync = harden(
+  (target, prop, methodArgs = undefined) => {
+    if (Array.isArray(methodArgs)) {
+      if (prop === null) {
+        // Function application.
+        return target(...methodArgs);
+      }
+      // Method application.
+      return target[prop](...methodArgs);
+    }
+    // Property get.
+    return target[prop];
+  },
+);
+
+const readOnlyProxyHandler = {
+  set(_target, _prop, _value) {
+    return false;
+  },
+  isExtensible(_target) {
+    return false;
+  },
+  setPrototypeOf(_target, _value) {
+    return false;
+  },
+  deleteProperty(_target, _prop) {
+    return false;
+  },
+};
+
+/**
+ * A Proxy handler for Sync(x)
+ *
+ * @param {*} x Any value passed to Sync(x)
+ * @param {GetApplySync} getApplySync
+ * @returns {ProxyHandler}
+ */
+function SyncProxyHandler(x, getApplySync) {
+  return harden({
+    ...readOnlyProxyHandler,
+    get(_target, p, _receiver) {
+      return (...args) => getApplySync(x, p, args);
+    },
+    apply(_target, _thisArg, argArray = []) {
+      return getApplySync(x, null, argArray);
+    },
+    has(_target, _p) {
+      // We just pretend everything exists.
+      return true;
```

Comment:
Seems ad-hoc. Is there a rationale for not doing alternatives like throwing an error?

Reply:
Yes.  Bob is forwarding properties to Alice's side, and we deliberately do not describe to Bob what the shape of Alice's objects are.